### ISSUE TITLE
Configuration of a rule - RF04

### ISSUE THREAD

**Issue by moseleyi** [sqlfluff/sqlfluff#6185]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

I'm trying to add some words to be ignored when evaluating aliases. This is what I'm doing:

```
[sqlfluff:rules:RF04]
ignore_words = week, month, quarter
```

But I'm getting this error:

```
WARNING    Rule configuration contain a section for unexpected rule 'RF04'. These values will be ignored. 
WARNING    The reference was however found as a match for rule RF04 with name 'references.keywords'. SQLFluff assumes configuration for this rule will be specified in 'sqlfluff:rules:references.keywords'. 
```

So then I change to this:
```
[sqlfluff:rules:references:keywords] 
ignore_words = week, month, quarter
```

And then I'm getting this error:
```
WARNING    Rule configuration contain a section for unexpected rule 'references'. These values will be ignored. 
WARNING    The reference was found as a match for multiple rules: {'RF03', 'RF02', 'RF05', 'RF01', 'RF04', 'RF06'}. Config should be specified by the name of the relevant rule e.g. 'sqlfluff:rules:capitalisation.keywords'. 
```

Now I'm completely confused how this rule is supposed to be configured

### Expected Behaviour

The words in my list are ignored and file passes linting

### Observed Behaviour

I get errors:

```
$ sqlfluff lint models/transformations/test.sql 
=== [dbt templater] Sorting Nodes...
=== [dbt templater] Compiling dbt project...
=== [dbt templater] Project Compiled.
WARNING    Rule configuration contain a section for unexpected rule 'references'. These values will be ignored. 
WARNING    The reference was found as a match for multiple rules: {'RF03', 'RF01', 'RF05', 'RF04', 'RF02', 'RF06'}. Config should be specified by the name of the relevant rule e.g. 'sqlfluff:rules:capitalisation.keywords'. 
== [models\transformations\test.sql] FAIL
L:   2 | P:  13 | RF04 | Keywords should not be used as identifiers.
                       | [references.keywords] 
All Finished 📜 🎉!
```

### How to reproduce

```sql
SELECT
  'test' AS week
```

### Dialect

dbt

### Version

3.1.0

### Configuration

```
[sqlfluff]
templater = dbt
dialect = bigquery
exclude_rules = RF02, AL01, AL05, ST07, AL03 
max_line_length = 150

[sqlfluff:templater:dbt]
project_dir = ./

[sqlfluff:indentation]
tab_space_size = 2
indented_joins = true
indented_using_on = false

[sqlfluff:rules]
custom_rules = custom_rules::Rule_Element_LL99

[sqlfluff:rules:capitalisation.keywords]
capitalisation_policy = upper

[sqlfluff:rules:capitalisation.identifiers]
extended_capitalisation_policy = lower

[sqlfluff:rules:capitalisation.functions]
extended_capitalisation_policy = upper

[sqlfluff:rules:capitalisation.literals]
capitalisation_policy = upper

[sqlfluff:rules:capitalisation.types]
extended_capitalisation_policy = upper

[sqlfluff:rules:aliasing.length]
min_alias_length = 3

[sqlfluff:rules:references:keywords] 
ignore_words = week, month, quarter
```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
